versioning/versioning.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+
+GIT_VERSION_COMMAND='git describe --tags --abbrev=8 --dirty 2> /dev/null'
+# Based on https://semver.org/#semantic-versioning-200 but allows `v` in front to match git tags
+GIT_SEMVER_REGEX=/^v?(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
+
+class Versioning
+  class << self
+    def current_version
+      verify_git!
+      verify_semver_tags!
+      version=`#{GIT_VERSION_COMMAND}`.strip
```

Comment:
I continue to believe that you should pass the version to `verify_semver_tag!` and not let it re-compute it itself internally. Calling the same external command twice seems wrong, and the code can diverge in the future, so the check could be against a different string than the one we return here:

```suggestion
      version=`#{GIT_VERSION_COMMAND}`.strip
      verify_semver_tag!(version)
```